Adyen.EcommLibrary/HttpClientHandler/HttpURLConnectionClient.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Adyen.EcommLibrary.Constants;
+using Adyen.EcommLibrary.HttpClientHandler.Interfaces;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Adyen.EcommLibrary.HttpClientHandler
+{
+    public class HttpUrlConnectionClient : IClient
+    {
+        public string Request(string endpoint, string json, Config config, bool isApiKeyRequired)
+        {
+            string responseText;
+
+            var httpWebRequest = (HttpWebRequest) WebRequest.Create(endpoint);
+            httpWebRequest.Method = "POST";
+            httpWebRequest.ContentType = "application/json";
+            //Set security protocol. Only TLS1.2
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+
+            //Use one of two authentication method.
+            if (isApiKeyRequired || !string.IsNullOrEmpty(config.XApiKey))
+            {
+                AddHeaders(config, httpWebRequest);
+            }
+            else
+            {
+                CreateBasicAuthentication(config, httpWebRequest);
```

Comment:
AddHeaders adds generic headers + x-api-key
shouldn't we add the generic headers anyway and just use this if-else for "x-api-key" or "Authorization" header?